src/Tribe/Views/V2/Widgets/Widget_Abstract.php

```diff
@@ -40,6 +40,61 @@ abstract class Widget_Abstract extends \Tribe\Widget\Widget_Abstract {
 	 */
 	protected $view_slug;
 
+	/**
+	 * @todo update in TEC-3612 & TEC-3613
+	 *
+	 * {@inheritDoc}
+	 * {@inheritDoc}default
+	 */
+	public function setup() {
+		// Add the admin template class for the widget admin form.
+		$this->admin_template = tribe( Admin_Template::class );
+
+		// Setup the View for the frontend.
+		$this->setup_view();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function form( $instance ) {
+
+		add_filter(
+			"tribe_widget_{$this->get_registration_slug()}_arguments",
+			function ( array $arguments ) use ( $instance ) {
+				return wp_parse_args(
+					$instance,
+					$arguments
+				);
+		} );
+
+		$arguments = $this->get_arguments();
+
+		$this->admin_template->template( 'widgets/list', $arguments );
+	}
+
+	/**
+	 * The function for saving widget updates in the admin section.
+	 *
+	 * @param array $new_instance
+	 * @param array $old_instance
+	 *
+	 * @return array The new widget settings.
+	 */
+	public function update( $new_instance, $old_instance ) {
+		$instance = $old_instance;
+
+		// todo update method so that is can be used in all the widgets.
+		/* Strip tags (if needed) and update the widget settings. */
+		$instance['title']                = strip_tags( $new_instance['title'] );
```

Comment:
:warning: `strip_tags()` does not strip CSS and JS in between the script and style tags. Use `wp_strip_all_tags()` to strip all tags.

<sub>hash: ffae7a2444501c9b5a52abd803cdc27d</sub>